### ISSUE TITLE
XPath – text() should use normalization

### ISSUE THREAD

**Issue by FilipJirsak** [dom4j/dom4j#21]
Input document can have one element divided into multiple text nodes (`text()` method than returns multiple nodes). XPath matching than doesn't work - it matches text only with first text node. Text nodes should be normalized before matching XPath.
Input text is divided into multiple text nodes for example when it contains entities - for example `&lt;` or `&gt;`.

**Comment by reluxa** [comment]
I guess it's belong to here: I was also trying to extract text() element form an xml document. the element looked liked the following

`<element>TOOXYZ:sometext /TOOXYZ:otherText</element>`

The above XML fragment is present many times in the xml document which I was trying to process. Interestingly the last occurrence of the XML could not get parsed correctly the text() has returned to nodes: `"TOOXYZ:sometext /TOO"` and `XYZ:otherText`

**Comment by FilipJirsak** [comment]
@reluxa This is correct, there can be multiple adjacent text nodes. You can call `document.normalize()` to settle text nodes.
This issue is about XPath matching - normalization should probably be done automatically before XPath matching because XPath expects normalized documents.

**Comment by ATorsius** [comment]
> This issue is about XPath matching - normalization should probably be done automatically before XPath matching because XPath expects normalized documents.

If XPath expects a normalized document, it should throw an error if the document isn't, instead of returning incorrect results.
